io/disk/arcconf/arcconf_raid_oper.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2016 IBM
+# Author: Manvanthara B Puttashankar <manvanth@linux.vnet.ibm.com>
+
+
+"""
+arcconf - Array configuration utility for PMC-Sierra
+(Microsemi) Controllers.
+"""
+
+import subprocess
+import time
+from avocado import Test
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import process
+from avocado import main
+
+
+class Arcconftest(Test):
+
+    """
+    arcconf_raid_oper.py covers functionality like RAID
+    creation, deletion, Physical disk & logical disk format,
+    and gather controller status using getconfig.
+    """
+
+    def setUp(self):
+
+        """
+        Checking if the required packages are installed,
+        if not found specific packages will be installed.
+        """
+
+        smm = SoftwareManager()
+        if not smm.check_installed("arcconf"):
+            self.log.info("arcconf should be installed prior to the test")
+            if smm.install("arcconf") is False:
+                self.skip("Unable to install arcconf")
+        self.crtl_no = self.params.get('crtl_no')
+        self.channel_no = self.params.get('channel_no')
+        self.disk_no = self.params.get('disk_no', default="").split(",")
+        self.pci_id = self.params.get('pci_id', default="").split(",")
+        self.logicaldrive = self.params.get('logicaldrive')
+        self.diskoverwrite = self.params.get('diskoverwrite')
+        self.disk_count = len(self.disk_no)
+        # Gets the list of PCIIDs on the system
+        cmd = 'for device in $(lspci | awk \'{print $1}\') ; do echo \
+              $(lspci -vmm -nn -s $device | grep "\[" | awk \'{print $NF}\' \
+              | sed -e "s/\]//g" | sed -e "s/\[//g" | tr \'\\n\' \' \' \
+              | awk \'{print $2,$3,$4,$5}\') ; done'
+
+        pci_id = self.cmdop_list(cmd, 'Y')
+        pci_id_formatted = []
+        pci_host_raw = self.cmdop_list('lspci -nn', 'Y')
+        pci_host = []
+
+        for i in range(0, len(pci_id)):
+            pci_id_formatted.append(str(pci_id[i]).replace(" ", ":"))
+            pci_host.append(str(pci_host_raw[i].replace(
+                pci_id_formatted[i][:8], pci_id_formatted[i])))
+        # check if all the yaml parameters are entered
+        if self.crtl_no is '' or self.channel_no is '' or self.disk_no is \
+           '' or self.pci_id is '' or self.disk_count <= 1:
```

Comment:
can this be not like this 
`
 '' or self.pci_id is '' or len(self.disk_no) <= 1:` just eliminate extra variable